src/block-components/icon/attributes.js

```diff
@@ -161,6 +161,20 @@ export const addAttributes = ( attrObject, options = {} ) => {
 		)
 	}
 
+	if ( ! hasIconGap ) {
+		finalExcludedAttributes.push(
+			'iconGap'
+		)
+	}
+
+	if ( ! hasIconPosition ) {
+		finalExcludedAttributes.push(
+			'iconPosition'
+		)
+	}
+
+	const iconAttributes = createIconAttributes( options )
+
```

Comment:
It's okay to include all the attributes since it won't have any effect in the block (except if there are any conflicts)